src/xcoobee/sdk/Consents.js

```diff
@@ -539,48 +544,22 @@ class Consents {
       apiUrlRoot,
     } = sdkCfg;
 
-    const errors = [];
-    const progress = [];
-    let response;
+    const apiAccessToken = await this._.apiAccessTokenCache.get(apiUrlRoot, apiKey, apiSecret);
+    const user = await this._.usersCache.get(apiUrlRoot, apiKey, apiSecret);
+    const userCursor = user.cursor;
 
-    try {
-      const apiAccessToken = await this._.apiAccessTokenCache.get(apiUrlRoot, apiKey, apiSecret);
-      const user = await this._.usersCache.get(apiUrlRoot, apiKey, apiSecret);
-      const userCursor = user.cursor;
+    if (filename) {
+      const endPointName = UploadPolicyIntents.OUTBOX;
+      const [uploadedFile] = await FileUtils.upload(apiUrlRoot, apiAccessToken, userCursor, endPointName, [filename]);
 
-      const successfullyUploadedFiles = [];
-      const result = { progress, ref_id: null };
-      if (filename) {
-        const endPointName = UploadPolicyIntents.OUTBOX;
-        const fileUploadResults = await FileUtils.upload(apiUrlRoot, apiAccessToken, userCursor, endPointName, [filename]);
-
-        fileUploadResults.forEach((fileUploadResult) => {
-          const { error, file, success } = fileUploadResult;
-          if (success) {
-            successfullyUploadedFiles.push(file);
-            progress.push(`successfully uploaded ${file}`);
-          } else {
-            errors.push(`Failed to upload file: ${file}. Error: ${error}.`);
-            progress.push(`failed upload on ${file}`);
-          }
-        });
-      }
-
-      if (!errors.length) {
-        const refId = await ConsentsApi.registerConsents(apiUrlRoot, apiAccessToken, resolvedCampaignId, targets, reference, successfullyUploadedFiles.length > 0 ? filename : null);
-        progress.push('successfully sent data response');
-        result.ref_id = refId;
+      if (!uploadedFile) {

```

Comment:
Because I don't see any change for `FileUtils.js`, I assume it is still behaving the same as before. So, it should be returning an array with a "file upload result" object for each file. Here, we are just checking the existence of the result object. I think this check will almost always be true. (I don't recall if `FileUtils.upload` will return a null result object in certain cases.) I think we also need to examine the contents of the result and check the `success` flag and the `error` prop.

Reply:
`uploadedFile` is not a result, its a first item of the result
```
      const [uploadedFile] = await FileUtils.upload(apiUrlRoot, apiAccessToken, userCursor, endPointName, [filename]);
```